cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -317,6 +317,54 @@ For example, give these arguments when starting the MapReduce through a RESTful
     "dataset.totals.output.partition.key.league" : "nfl"
   }
 
+Incrementally Processing Partitions of a PartitionedFileSets in MapReduce
+=========================================================================
+
+Another way to process a partitioned file set from a MapReduce is via the beta feature of incrementally
+consuming its partitions. Using this, a MapReduce job can run and process the files of all partitions
+committed to a particular partitioned file set since the last run of the MapReduce job. Currently, the
+MapReduce job utilizing this feature is responsible for storing its own state of processing. The following
+example demonstrates how to use the ``BatchPartitionConsumer`` class. Note that it writes and reads the
+state from a particular row in a KeyValueTable, but other types Datasets can also be used for persistence::
+
+  public static class WordCount extends AbstractMapReduce {
+
+    private final BatchPartitionConsumer batchPartitionConsumer = new BatchPartitionConsumer() {
+      private static final String STATE_KEY = "state.key";
+
+      @Nullable
+      @Override
+      protected byte[] readBytes(DatasetContext datasetContext) {
+        return ((KeyValueTable) datasetContext.getDataset("consumingState")).read(STATE_KEY);
+      }
+
+      @Override
+      protected void writeBytes(DatasetContext datasetContext, byte[] stateBytes) {
+        ((KeyValueTable) datasetContext.getDataset("consumingState")).write(STATE_KEY, stateBytes);
+      }
+    };
+
+Then, in the ``beforeSubmit()`` method of the MapReduce, specify the partitioned file set to be used as input::
```

Comment:
I think it does not become clear that the "getPartitionedFileSet()" does two things (at least):
- read back the previous state
- determine the new partitions to read 
- modify (but not persist) the state 
- instantiate the dataset with the discovered partitions as input. 
I think you can write more prose around that. Also consider renaming the method (which looks like a simple getter) right now